src/value.cc

```diff
@@ -383,6 +413,9 @@ class tostring_visitor : public boost::static_visitor<std::string>
     return (boost::format("[%1% : %2% : %3%]") % v.begin_val % v.step_val % v.end_val).str();
   }
 
+  std::string operator()(const std::shared_ptr<Expression> &v) const {
+	  return "[function]";
```

Comment:
Would it have any merit to emit some unique identifier in addition?

Reply:
Not pushed yet, but I've changed that to print the actual function definition. I guess that's needed once we continue with object-literals :-) (which I'm playing with to see how that interacts with the function-literals).

Reply:
Pushed the change to print the actual function definition. Will revisit when converting to a specific `FunctionLiteral` object instead of generic `Expression`.